runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/DoFnOperator.java

```diff
@@ -538,9 +565,28 @@ public void onProcessingTime(InternalTimer<Object, TimerData> timer) throws Exce
     fireTimer(timer);
   }
 
+  /**
+   *  StateNamespace don't have a getWindow method.
+   */
+  private BoundedWindow getWindowFromNamespace(StateNamespace namespace) {
```

Comment:
Here or in `fireTimer`, if these are really only user timers, that you can just `checkArgument(namespaces instanceof WindowNamespace)`.